src/lib/geo-location.js

```diff
@@ -56,11 +55,18 @@ const getGeoLocation = async ({ mode, offline, state }) => {
     }
   }
 
-  // If the `--geo` option is set to `mock`, we use the mock location. Also,
-  // if the `--offline` option was used, we can't talk to the API, so let's
-  // also use the mock location.
-  if (mode === 'mock' || offline) {
-    return mockLocation
+  // If the `--geo` option is set to `mock`, we use the default mock location.
+  // If the `--offline` option was used, we can't talk to the API, so let's
+  // also use the mock location.  Otherwise, use the country code passed in by
+  // the user.
+  if (!['cache', 'update'].includes(mode) || offline) {
+    return mode === 'mock' && !geoCountry
```

Comment:
Personally, I find this a bit hard to parse. Consider moving this to separate return statements.

```js
if (mode === 'mock') {
  if (geoCountry) {
    return { ... }
  }

  return mockLocation
}
```